src/AllowedField.php

```
@@ -3,28 +3,38 @@
+    public static function setFilterArrayValueDelimiter(?string $delimiter = null): void
     {
         if (isset($delimiter)) {
             QueryBuilderRequest::setFilterArrayValueDelimiter($delimiter);
         }
     }
 
-    public static function partial(string $name, $internalNames = null, bool $addRelationConstraint = true, string $arrayValueDelimiter = null): self
+    /**
+     * @param  string|array<int, string>|null  $internalNames
+     */
+    public static function partial(string $name, string|array|null $internalNames = null, bool $addRelationConstraint = true, ?string $arrayValueDelimiter = null): self
```

PHPMD (2.15.0)

[error] The method partial has a boolean flag argument $addRelationConstraint, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)